New `snapshot_meta_column_names` config for dbt snapshots (#6211)
fishtown-analytics/docs.getdbt.com: website/docs/reference/resource-configs/snapshot_meta_column_names.md

---
resource_types: [snapshots]
description: "Snapshot meta column names"
datatype: "{<dictionary>}"
default_value: {"dbt_valid_from": "dbt_valid_from", "dbt_valid_to": "dbt_valid_to", "dbt_scd_id": "dbt_scd_id", "dbt_updated_at": "dbt_updated_at"}
id: "snapshot_meta_column_names"
---

> Available in v1.9 or with [versionless](/docs/dbt-versions/upgrade-dbt-version-in-cloud#versionless) dbt Cloud.

Reviewer: starting in 1.9?

Reviewer: do we want to make this page versioned? (only be available for 1.9+)

Reviewer: <img width="661" alt="Screenshot 2024-10-02 at 4 51 48 PM" src="https://github.com/user-attachments/assets/7fc7baae-15b1-434c-9e70-d4021ddb5e12">

nvm i see that you've already done this!

Author: I did it based on your suggestion -- good call 🙌

      dbt_updated_at: <string>

```

</File>

## Description

In order to align with an organization's naming conventions, the `snapshot_meta_column_names` config can be used to customize the names of the [metadata columns](/docs/build/snapshots#snapshot-meta-fields) within each snapshot.

## Default

By default, dbt snapshots use the following column names to track change history using [Type 2 slowly changing dimension](https://en.wikipedia.org/wiki/Slowly_changing_dimension#Type_2:_add_new_row) records:

| Field          | Meaning | Notes |
| -------------- | ------- | ----- |
| `dbt_valid_from` | The timestamp when this snapshot row was first inserted and became valid. | The value is affected by the [`strategy`](/reference/resource-configs/strategy). |
| `dbt_valid_to`   | The timestamp when this row is no longer valid. |  |
| `dbt_scd_id`     | A unique key generated for each snapshot row. | This is used internally by dbt. |
| `dbt_updated_at` | The `updated_at` timestamp of the source record when this snapshot row was inserted. | This is used internally by dbt. |

However, these column names can be customized using the `snapshot_meta_column_names` config.

> :::note

Reviewer 2: We might want to use a warning as this could be really annoying for them if they don't see it.
```suggestion
:::warning
```

> To avoid any uninentional data modification, dbt will **not** automatically apply any column renames. So if a user applyies `snapshot_meta_column_names` config for a snapshot without updating the pre-existing table, they will get an error. Our recommendation is to either only use these settings for net-new snapshots, or to arrange an update of pre-existing tables prior to committing a column name change.

Reviewer 2: ```suggestion
To avoid any unintentional data modification, dbt will **not** automatically apply any column renames. So if a user applies `snapshot_meta_column_names` config for a snapshot without updating the pre-existing table, they will get an error. We recommend either only using these settings for net-new snapshots, or arranging an update of pre-existing tables prior to committing a column name change.
```

Author: holi cow i uninentionallly speled every word wrongly -- thank you for fixing this up! 🧠
